### ISSUE TITLE
Support for multiple "done" values in egg startup configurations

### ISSUE THREAD

**Issue by matthewpi** [pterodactyl/panel#2220]
**Is your feature request related to a problem? Please describe.**
Some eggs might require multiple `done` values depending on Server Variables to ensure the server gets marked as online.

**Describe the solution you'd like**
Either support for `done` to be an array in the startup configuration or a split character in the existing done string, such as `|`.

I understand that some people might be concerned about a split character accidentally breaking existing eggs so we could also add a `split_character` option which if set determines how the `done` option would be split, allowing different eggs to use different split characters to avoid interfering with characters that need to be matched.

**Describe alternatives you've considered**
None

**Comment by parkervcp** [comment]
Honestly making it a variable would be the best.

**Comment by DaneEveritt** [comment]
An array is probably fine, really easy to maintain backwards compat by just checking if it is a string or array. I also wanted to add support for regex matching by prefixing the string with like `regex:`.

**Comment by ririko5834** [comment]
Is this added? Can I provide array of done keywords? I think that you should make it that users can edit it per server.

**Comment by DaneEveritt** [comment]
I'm not adding it per server, and it is done as indicated by the line:

```
DaneEveritt closed this in pterodactyl/wings#47 on Aug 4, 2020
```

**Comment by ririko5834** [comment]
Ok, the array is fine
